packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -216,6 +213,11 @@ export class GarbageCollector implements IGarbageCollector {
 					return undefined;
 				}
 
+				// If the GC Version isn't current, don't load the GC data - we'll regenerate it anyway.
+				if (this.configs.gcVersionInBaseSnapshot !== this.configs.gcVersionInEffect) {
+					return undefined;
+				}
+
```

Comment:
@agarwal-navin this is the only change, the rest is #14657 (which isn't yet merged).  There's also a new test.

Reply:
I was thinking that this check should be done in initializeBaseState and this.initializeGCStateFromBaseSnapshotP but I think this is better. Basically, we don't use this data at all.